indexer/src/processing/events.processing.ts

```diff
@@ -0,0 +1,196 @@
+import { ConfigChangedHandler } from './nft/config-changed.handler';
+import { ImageChangedHandler } from './nft/image-changed.handler';
+import { TransferredHandler } from './nft/transferred.handler';
+import { MetadataAddedHandler } from './nft/metadata-added.handler';
+import { EventInfo } from './event-info.type';
+import { EntitiesService } from './entities.service';
+
+const marketplaceMeta = ProgramMetadata.from(
+  readFileSync('./assets/nft_marketplace.meta.txt', 'utf8'),
+);
+const nftMeta = ProgramMetadata.from(
+  readFileSync('./assets/nft.meta.txt', 'utf8'),
+);
+
+const marketplaceEventsToHandler: Record<
+  NftMarketplaceEventType,
+  INftMarketplaceEventHandler | undefined
+> = {
+  [NftMarketplaceEventType.NewCollectionAdded]: new NewCollectionAddedHandler(),
+  [NftMarketplaceEventType.CollectionCreated]: new CollectionCreatedHandler(),
+  [NftMarketplaceEventType.CollectionDeleted]: new CollectionDeletedHandler(),
+
+  [NftMarketplaceEventType.SaleNft]: new SaleNftHandler(),
+  [NftMarketplaceEventType.SaleNftCanceled]: new SaleNftCanceledHandler(),
+  [NftMarketplaceEventType.NftSold]: new NftSoldHandler(),
+
+  [NftMarketplaceEventType.AdminDeleted]: new AdminDeletedHandler(),
+  [NftMarketplaceEventType.AdminsAdded]: new AdminAddedHandler(),
+
+  [NftMarketplaceEventType.AuctionCreated]: new AuctionCreatedHandler(),
+  [NftMarketplaceEventType.BidAdded]: new BidAddedHandler(),
+  [NftMarketplaceEventType.AuctionClosed]: new AuctionClosedHandler(),
+  [NftMarketplaceEventType.AuctionCanceled]: new AuctionCanceledHandler(),
+
+  [NftMarketplaceEventType.OfferCreated]: new OfferCreatedHandler(),
+  [NftMarketplaceEventType.OfferCanceled]: new OfferCanceledHandler(),
+  [NftMarketplaceEventType.OfferAccepted]: new OfferAcceptedHandler(),
+
+  [NftMarketplaceEventType.ConfigUpdated]: new ConfigUpdatedHandler(),
+};
+
+const nftEventsToHandler: Record<NftEventType, INftEventHandler | undefined> = {
+  [NftEventType.Initialized]: new InitializedHandler(),
+  [NftEventType.Minted]: new NftMintedHandler(),
+  [NftEventType.Approved]: new NftApprovedHandler(),
+  [NftEventType.ApprovalRevoked]: new NftApprovalRevokedHandler(),
+  [NftEventType.Expanded]: undefined,
+  [NftEventType.ConfigChanged]: new ConfigChangedHandler(),
+  [NftEventType.ImageChanged]: new ImageChangedHandler(),
+  [NftEventType.MetadataAdded]: new MetadataAddedHandler(),
+  [NftEventType.TokenInfoReceived]: undefined,
+  [NftEventType.Transferred]: new TransferredHandler(),
+};
+
+export class EventsProcessing {
+  constructor(private readonly entitiesService: EntitiesService) {}
+
+  saveAll() {
+    return this.entitiesService.saveAll();
+  }
+
+  async handleMarketplaceEvent(
+    payload: string,
+    eventInfo: EventInfo,
+  ): Promise<NftMarketplaceEvent | null> {
+    const { blockNumber, txHash } = eventInfo;
+    try {
+      console.log(`${blockNumber}-${txHash}: handling marketplace event`);
```

Comment:
you can't get anything from this log. TxHash always be a hash of `gear.run` extrinsic that can contain a lot of different messages. It's better to have `message.id` in this log